src/App.tsx

```diff
@@ -15,47 +16,130 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Alphabetically = 'name',
+  ByLength = 'length',
+}
+
+enum SortOrder {
+  Default = 0,
+  Ascending = 1,
+  Descending = -1,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  isReversed: SortOrder,
+): string[] {
+  const preparedGoods: string[] = [...goods];
+
+  if (sortField !== SortType.Default) {
+    preparedGoods.sort((firstItem, secondItem) => {
+      if (sortField === SortType.Alphabetically) {
+        return firstItem.localeCompare(secondItem);
+      }
+
+      if (sortField === SortType.ByLength) {
+        return firstItem.length - secondItem.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (isReversed === SortOrder.Ascending) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(SortOrder.Default);
+  const isResetButtonVisible = sortField !== SortType.Default
+    || isReversed === SortOrder.Ascending;
```

Comment:
```suggestion
  const isResetButtonVisible = sortField || isReversed;
```

Reply:
My app doesn't work the way you advised because
enum has 3 options